pkg/resources/apiserver/deployment.go

```diff
@@ -357,7 +385,11 @@ func getApiserverFlags(data *resources.TemplateData, etcdEndpoints []string, ena
 	if cluster.Spec.Cloud.GCP != nil {
 		flags = append(flags, "--kubelet-preferred-address-types", "InternalIP")
 	} else {
-		flags = append(flags, "--kubelet-preferred-address-types", "ExternalIP,InternalIP")
+		if data.IsKonnectivityEnabled() {
+			flags = append(flags, "--kubelet-preferred-address-types", "InternalIP,ExternalIP")
```

Comment:
Why exactly is this needed and what are the potential outcomes / risks?

Reply:
> Why... 

Without this fetching logs for pods in user-cluster fails. The problem discussed [here](https://kubermatic.slack.com/archives/C01EWQZEW69/p1628769575001400). 

KAS tries to connect to kubelet via konnectivity-agent in the user-cluster. This request fails because of security policies disallow external traffic to the node.  

> risks... 

All egress traffic from KAS goes through the konnectivity-proxy and to the konnectivity-agents.  Agents are authenticated when they register themselves with konenctivity-proxy. 

Without this we had to modify current security policy of the node to allow external traffic. 

So no security issues that I can spot. 

cc @youssefazrak @irozzo-1A

Reply:
OK, understood, thx for the explanation.. maybe a comment in there would help others who will be reading the code later.

Reply:
I confirm, and agree with @rastislavs that a comment here could result useful.